File: services/_future/llm-gateway/src/controllers/health_controller.py
"""Health check controller."""

from fastapi import APIRouter, Request

router = APIRouter()


@router.get("/")
async def health_check(request: Request):
    checks = {}
    try:
        await request.app.state.redis.ping()
        checks["redis"] = "healthy"
    except Exception:
        checks["redis"] = "unhealthy"
    is_healthy = all(v == "healthy" for v in checks.values())
    return {"status": "healthy" if is_healthy else "degraded", "service": "llm-gateway", "checks": checks}

File: services/_future/auth-service/src/controllers/healthController.js
'use strict';

async function healthRoutes(fastify) {
    fastify.get('/', async (request, reply) => {
        const checks = {};

        // Database check
        try {
            await fastify.db.raw('SELECT 1');
            checks.database = 'healthy';
        } catch {
            checks.database = 'unhealthy';
        }

        // Redis check
        try {
            await fastify.redis.ping();
            checks.redis = 'healthy';
        } catch {
            checks.redis = 'unhealthy';
        }

        const isHealthy = Object.values(checks).every((s) => s === 'healthy');

        return reply.status(isHealthy ? 200 : 503).send({
            status: isHealthy ? 'healthy' : 'degraded',
            service: 'auth-service',
            timestamp: new Date().toISOString(),
            checks,
        });
    });
}

module.exports = healthRoutes;
